src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -268,13 +277,57 @@ private async Task RunStepsAsync(List<IStep> embeddedSteps)
                     step.ExecutionContext.Complete(TaskResult.Failed);
                 }
 
-                await RunStepAsync(step);
+                // Evaluate condition
+                step.ExecutionContext.Debug($"Evaluating condition for step: '{step.DisplayName}'");
```

Comment:
This mirrors the Steps Runner functionality: https://github.com/actions/runner/blob/4e95d0d6ad46e454749cc5d69a7072df34a8e12f/src/Runner.Worker/StepsRunner.cs#L181

Reply:
cool, does this mean we support condition on steps now?

Reply:
i dont think this is complete. We need to register the callback so we can re-test when cancellation actually fires. This just tests before the step is run. We need to re-test when cancellation fires and the step is in-progress.

Reply:
Not at the moment, but the infrastructure is there we just need to enable the keyword (and be sure its the functionality we actually want)

Reply:
if this PR is incrementally correct, you might want to figure out the callback thing in a future PR

Reply:
...since this feature is behind a FF anyway

Reply:
I went ahead and put it in this pr, the changes are relatively scoped and was able to do testing for it